src/components/tooltip/CallNode.js

```diff
@@ -57,15 +62,241 @@ type Props = {|
+          return;
+        }
+        const categoryName = categories[category].name;
+        categoriesAndTime.push({
+          categoryName,
+          category,
+          subCategory: -1,
+          selfTime: selfTimeValue,
+          totalTime: entireCategoryValue,
+        });
+        if (filteredSubcategoryBreakdown[0].subcategory === 0) {
+          const other = filteredSubcategoryBreakdown[0];
+          filteredSubcategoryBreakdown.pop();
+          filteredSubcategoryBreakdown.push(other);
+        }
+        filteredSubcategoryBreakdown.forEach((sub) => {
+          const { self, subcategory } = sub;
+          const selfTimeValue = selfTime.breakdownByCategory
+            ? selfTime.breakdownByCategory[category].subcategoryBreakdown[
+                subcategory
+              ]
+            : 0;
+          categoriesAndTime.push({
+            categoryName,
+            category,
+            subCategory: subcategory,
+            selfTime: selfTimeValue,
+            totalTime: self,
+          });
+        });
+      }
+    );
+
+    return (
+      <div className="tooltipCallNodeCategory">
+        {/* grid row -------------------------------------------------- */}
+        <div />
+        <div className="tooltipCallNodeHeader" />
+        <div className="tooltipCallNodeHeader">
+          <span className="tooltipCallNodeHeaderSwatchRunning" />
+          Running
+        </div>
+        <div className="tooltipCallNodeHeader">
+          <span className="tooltipCallNodeHeaderSwatchSelf" />
+          Self
+        </div>
+        {/* grid row -------------------------------------------------- */}
+        <div className="tooltipLabel tooltipCategoryLabel tooltipCategoryRowHeader">
+          Overall
+        </div>
+        <div className="tooltipCallNodeGraph tooltipCategoryRow tooltipCategoryRowHeader">
+          <div
+            className="tooltipCallNodeCategoryGraphRunning"
+            style={{
+              width: GRAPH_WIDTH,
+            }}
+          />
+          <div
+            className="tooltipCallNodeCategoryGraphSelf"
+            style={{
+              width: (GRAPH_WIDTH * selfTime.value) / totalTime.value,
+            }}
+          />
+        </div>
+        <div className="tooltipCallNodeTiming tooltipCategoryRow tooltipCategoryRowHeader">
+          {formatCallNodeNumberWithUnit(
+            weightType,
+            isHighPrecision,
+            totalTime.value
+          )}
+        </div>
+        <div className="tooltipCallNodeTiming tooltipCategoryRow tooltipCategoryRowHeader">
+          {selfTime.value === 0
+            ? '—'
+            : formatCallNodeNumberWithUnit(
+                weightType,
+                isHighPrecision,
+                selfTime.value
+              )}
+        </div>
+        {categoriesAndTime.map((entry, index) => {
+          const categoryColor = categories[entry.category].color;
+          return (
+            <React.Fragment key={index}>
```

Comment:
The goal for the `key` is that this uniquely identifies the line, so that React can know that a line's content doesn't change when the order changes. In this case maybe `entry.categoryName` would work fine?

Reply:
Of course.